common/scripted_triggers/wcsp_scripted_triggers.txt

```diff
@@ -1,39 +1,40 @@
 # Long potential's, allow's, trigger's, limit's are stored here
-ask_to_be_your_mentor_decision_from_potential = {
+prev_mentor_this_class_student_potential = {
 	is_untouchable_trigger = no
 
 	# is_adult = yes
 	is_incapable = no
 	prisoner = no
 
-	NOT =  { reverse_has_opinion_modifier = { modifier = opinion_teaching_me who = ROOT } }	# Isn't your mentor right now
-	OR = {
-		NOT = { has_character_flag = sent_letter_to_mentor }
-		had_character_flag = { flag = sent_letter_to_mentor years = 1 }
-	}
+	NOT =  { reverse_has_opinion_modifier = { modifier = opinion_teaching_me who = PREV } }	# Isn't your mentor right now
 	is_training_self_trigger = no
 	is_training_mentor_trigger = no
+	
+	# Evil characters can only train characters of their religion
```

Comment:
As I know, there's no way to hide the decision if there are no characters without hurting performance.

Reply:
Maybe instead of "if there are no characters", hide the decision if decision target is evil and character that press the decision don't have the same religion as decision target?

Reply:
Changed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/948/commits/e7041ec06017aaac5e48ae02f36346c7aa0826bb.